### PR TITLE
Sample code for the article on closures

### PR DESCRIPTION
**Pull request by lpozo**
**Where to put new files:**

- New files should go into a top-level subfolder, named after the article slug. For example: `my-awesome-article`

**How to merge your changes:** 

1. [Make sure the CI code style tests all pass (+ run the automatic code formatter if necessary).](https://github.com/realpython/materials/blob/master/README.md)
2. Find an RP Team member on Slack and ask them to review & approve your PR.
3. Once the PR has one positive ("approved") review, GitHub lets you merge the PR.
4. 🎉

### CODE REVIEW

File: python-closure/closure.py [hunk 1]
```diff
@@ -0,0 +1,16 @@
...
+
+#     def inner_func():
+#         print(f"Hello, {name}!")
+
+#     return inner_func
+
+
+def outer_func():
+    name = "Pythonista"
+    return lambda name=name: print(f"Hello, {name}!")
```
Comment (bzaczynski) [inline python-closure/closure.py:12]: :warning: **Fact Check**

@lpozo Actually, I just realized this is a wrong syntax for a lambda expression. This is the correct one:

```suggestion
    return lambda name: print(f"Hello, {name}!")
```

Unfortunately, this mistake is also present in the tutorial text.

**Review by bzaczynski — CHANGES_REQUESTED** [review]
@lpozo There's one small detail to fix in the lambda expression example (same in the tutorial text.)

File: python-closure/closure.py (hunk 1 above)
Comment (lpozo) [reply python-closure/closure.py:12]: It doesn't work if we don't set the argument:

```python
>>> def outer_func():
...     name = "Pythonista"
...     return lambda name: print(f"Hello, {name}!")
...     
... 
>>> func = outer_func()
>>> func()
Traceback (most recent call last):
  File "<input>", line 1, in <module>
    func()
TypeError: outer_func.<locals>.<lambda>() missing 1 required positional argument: 'func'
```

Comment (lpozo) [reply python-closure/closure.py:12]: The correct syntax is:

```python
>>> def outer_func():
...     name = "Pythonista"
...     return lambda: print(f"Hello, {name}!")
...     
... 
>>> func = outer_func()
>>> func()
Hello, Pythonista!
```

**Review by bzaczynski — APPROVED** [review]

**Review by brendaweles — APPROVED** [review]

**Pull request merged**